packages/frontend/src/components/delete_document_dialog.tsx

```diff
@@ -0,0 +1,114 @@
+import { Button } from "catcolab-ui-components";
+import { createSignal } from "solid-js";
+import { useApi } from "../api";
+import { Dialog } from "./dialog";
+
+export type DeleteDocumentInfo = {
+    refId: string;
+    name: string | null;
+    typeName: string;
+};
+
+function noop(_: boolean): void {}
+
+export function useDeleteDocument(docInfo: DeleteDocumentInfo) {
+    const api = useApi();
+
+    const [showDeleteConfirm, setShowDeleteConfirm] = createSignal(false);
+    const [showError, setShowError] = createSignal(false);
+    const [errorMessage, setErrorMessage] = createSignal("");
+    const [resolveDeletion, setResolveDeletion] = createSignal<(success: boolean) => void>(noop);
```

Comment:
Do we need `noop`? Seems clear to have

```typescript
    const [resolveDeletion, setResolveDeletion] = createSignal<(success: boolean) => void | undefined>();
```

and then call

```typescript
resolveDeletion()?.(...);
```

In fact, you're already doing the latter in some but not all places below.